eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/Field.php

```diff
@@ -154,11 +159,21 @@ public function handle(
         );
 
         $whereExpressions = array();
+
+        $columnInfoArray = array_column($fieldsInformation, 'column');
+
+        if (false === reset($columnInfoArray) && 1 === count(array_unique($columnInfoArray))) {
+            throw new NotImplementedException(
```

Comment:
This whole block looks rather complicated. Not that much, but wouldn't we achieve the same result checking if `$whereExpressions` is empty after the next loop?